openpuc_scrapers/scrapers/ny.py

```diff
@@ -285,8 +217,77 @@ def make_dedupe_string(file: NYPUCFileData) -> str:
     return list(return_vals)
 
 
-if __name__ == "__main__":
-    # For local testing
-    print("Running scraping workflow...")
-    results = full_scraping_workflow()
-    print(f"Scraped {len(results)} dockets with files")
+class NYPUCScraper(GenericScraper[NYPUCDocketInfo, NYPUCFileData]):
```

Comment:
@cappuccinocosmico can you explain the benefit of using a parameterized type here? (e.g. `GenericScraper[NYPUCDocketInfo, NYPUCFileData]`)